digits/extensions/data/imageProcessing/data.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) 2016, NVIDIA CORPORATION.  All rights reserved.
+from __future__ import absolute_import
+
+import math
+import os
+import random
+
+import numpy as np
+
+from digits.utils import image, subclass, override, constants
+from ..interface import DataIngestionInterface
+from .forms import DatasetForm
+
+TEMPLATE = "template.html"
+
+
+@subclass
+class DataIngestion(DataIngestionInterface):
+    """
+    A data ingestion extension for an image processing dataset
+    """
+
+    def __init__(self, **kwargs):
+        """
+        the parent __init__ method automatically populates this
+        instance with attributes from the form
+        """
+        super(DataIngestion, self).__init__(**kwargs)
+
+        self.random_indices = None
+
+        if not 'seed' in self.userdata:
+            # choose random seed and add to userdata so it gets persisted
+            self.userdata['seed'] = random.randint(0, 1000)
+
+        random.seed(self.userdata['seed'])
```

Comment:
Yes I need to save the seed here as I want to generate the same random indices from two separate processes (the process that creates the training DB and the process that creates the validation DB). Indeed this is different from the (many) other seeds that are used in the places you pointed at. Here a new seed is generated every time we create a dataset (however it's done only *once* per dataset).